ldms/src/core/ldms.h

```diff
@@ -1381,8 +1381,35 @@ void ldms_qgroup_info_free(ldms_qgroup_info_t qinfo);
 typedef enum ldms_stream_type_e {
 	LDMS_STREAM_STRING,
 	LDMS_STREAM_JSON,
+	LDMS_STREAM_AVRO_SER,
+	LDMS_STREAM_LAST, /* the last enumureation; not a real type */
 } ldms_stream_type_t;
 
+/**
+ * \brief Stream Type Symbol.
+ *
+ * This function returns a constant string symbol (e.g. "LDMS_STREAM_STRING") of
+ * the given Stream type \c t. If the given type is out of valid range,
+ * "UNKNOWN" is returned.
+ *
+ * \param t The Stream type.
+ *
+ * \retval s The symbol (e.g. "LDMS_STREAM_STRING").
+ */
+const char *ldms_stream_type_sym(ldms_stream_type_t t);
+
+typedef struct ldms_stream_client_s *ldms_stream_client_t;
+typedef struct json_entity_s *json_entity_t;
+
+/* from avro/schema.h */
+typedef struct avro_obj_t *avro_schema_t;
```

Comment:
Please, please don't do this. We should get this definition by doing #include <avro/schema.h>. Please don't cut-and-paste definitions from external packages.

I can only presume that this is done to avoid having ldms.h have ever-present dependencies on external avro packages. But the solution to this problem is to just introduce a separate header. Make building avro support optional if we want the dependency to be optional.

Reply:
@morrone your presumption is correct. I first did "ldms_avro.h" and "ldms_avro.h". It has only one API, so I just put it here as it is more convenient and I was thinking that this `typedef` doesn't care about the definition of the structure. But, if `avro` somehow changes redefine `avro_schema_t` to something else and we're done, which is bad.

Thanks for catching this, I'll put the avro bits into its own thing.